src/main/java/duke/DukeIO.java

```diff
@@ -0,0 +1,116 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
+public class DukeIO {
+
+    private static final String DUKEDATADIR = "data";
+    private static final String DUKEDATA = "data/dukeData.txt";
+
+    // create file and directory if they are not created yet
+    public static void initialize() throws IOException {
+        File dataDirectory = new File(DUKEDATADIR);
+        if (!dataDirectory.exists()){
+            dataDirectory.mkdirs();
+        }
+        File dukeData = new File(DUKEDATA);
+        if(!dukeData.exists()){
+            dukeData.createNewFile();
+        }
+    }
+
+    public static void writeDukeData() throws IOException {
+        // clear existing data in the txt file
+        File file = new File(DUKEDATA);
+        if(file.exists()){
+            file.delete();
+        }
+        file.createNewFile();
+        FileWriter fw = new FileWriter(DUKEDATA);
+
+        String line; //string to write to the txt file(per line)
+        String taskNum; //task number
+        int padding; //adding certain spaces after the task number
+        String taskIsDone;
+        String taskDescription;
+        Task[] tasks = DukeController.getTasks();
+        int currentTaskLength = DukeController.getCurrentTaskLength();
+
+        // format data line by line and write to the txt file
+        // example formatting for todo:     1   [T][1] todo eat
+        // example formatting for deadline: 35  [D][0] deadline assignment /by 29990930
+        // example formatting for event:    100 [E][1] event sleep /on 29991001
+        for(int i=0; i<currentTaskLength; i++){
+            //common formatting for the 3 task types
+            taskNum = String.valueOf(i+1);
+            padding = 3 - taskNum.length();
+            for(int j=0; j<padding; j++){
+                taskNum =  taskNum + " ";
+            }
+            if(tasks[i].getIsDone() == true){
+                taskIsDone = "[1] ";
+            }else{
+                taskIsDone = "[0] ";
+            }
+
+            //individual formatting for the 3 task types
+            if(tasks[i] instanceof ToDo){
+                taskDescription = tasks[i].getDescription();
+                line = taskNum + " [T]" + taskIsDone + taskDescription;
+            }else if(tasks[i] instanceof Deadline){
+                taskDescription = tasks[i].getDescription() + "/" + ((Deadline) tasks[i]).getKeyWordBeforeDeadline()
```

Comment:
If possible, try and break lines of code similar to these down into multiple intermediary lines of code, for enhanced code readability